hasher-matcher-actioner/hmalib/lambdas/api/submit.py

```diff
@@ -142,6 +161,9 @@ def submit(
         if request.submission_type == SubmissionType.UPLOAD.name:
             fileName = request.content_id
             fileContents = base64.b64decode(request.content_ref)
+
+            record_content_submission(dynamodb_table, image_folder_key, request)
```

Comment:
I suspect this is being done before s3_client.put_object to prevent a case where hasher starts to hash but content object doesn't exist yet. Document?